features/facebookImport/src/model/storage.js

```diff
@@ -37,15 +37,33 @@ export class ZipFile {
     constructor(file, pod) {
         this._pod = pod;
         this._file = file;
+
+        this._entriesList = null;
+        this._entriesSet = null;
     }
 
     get id() {
         return this._file.id;
     }
 
-    async getEntries() {
+    async _readEntriesList() {
         const { polyOut } = this._pod;
-        return polyOut.readdir(this.id);
+        return await polyOut.readdir(this.id);
+    }
+
+    async refreshCachedEntries() {
+        this._entriesList = await this._readEntriesList();
```

Comment:
I would suspect the order of entries is irrelevant, so I'd be all for having `getEntries` and `this._entries` be a `Set` in the first place. The code working with that can work with the set instead of needing an array, hm? Seems less cumbersome than introducing a dedicated `hasEntries` the caller needs to remember to use. If all they can do is `getEntries().has(...)`, it seems harder to accidentally do the O(n) lookup.

Reply:
One difference is that how we get the size: `size` vs `length`. Maybe it is easier now to store the list internally as a set and have `getEntries ` return an array and change that later on.